packages/next-tinacms-github/README.md

```diff
@@ -32,16 +49,94 @@ import { apiProxy } from 'next-tinacms-github'
 export default apiProxy
 ```
 
-# `previewHandler`
-Handles setting the preview data from Github cookies
+### `previewHandler`
 
-## Implementation
+Handles setting the the Nextjs [preview data](https://nextjs.org/docs/advanced-features/preview-mode) from your cookie data.
 
 ```
 // pages/api/preview.ts
 
 import { previewHandler } from 'next-tinacms-github'
 
 export default previewHandler
+```
+
+### Loading content from Github
+
+The `preview` data, which gets set by calling your [preview function](#previewhandler), will be accesible through `getStaticProps` throughout your app.
+
+```ts
+//Blog template [slug].ts
+
+import {
+  getMarkdownFile as getGithubMarkdownFile,
+} from 'next-tinacms-github'
+
+// ...
+
+export const getStaticProps: GetStaticProps = async function({
+  preview,
+  previewData,
+  ...ctx
+}) {
+  const { slug } = ctx.params
+
+  let file = {}
+  const filePath = `content/blog/${slug}.md`
+
+  const sourceProviderConnection = {
+    forkFullName: previewData.fork_full_name,
+    headBranch: previewData.head_branch || 'master', 
+  }
+  
+  let error = null
+
+  if(preview) {
+    try {
+    file = await getGithubMarkdownFile(filePath,
+      sourceProviderConnection, 
+      previewData.accessToken)
+    }
+    catch (e) {
+      // If there is an error initially loading the content from Github, we want to display an actionable error
+      // to the user. They may need to re-authenticate or create a new fork.
+      if (e instanceof GithubError) {
+        error = { ...e } //workaround since we cant return error as JSON
+      } else {
+        throw e
+      }
+    }
+  }
+  else {
+    // Get your production content here
+    // when you are not in edit-mode
+     file = await readLocalMarkdownFile(filePath)
+  }
+
+  return {
+    props: {
+      sourceProviderConnection,
+      editMode: !!preview,
+      file,
+      error
+    },
+  }
+}
+```
+
+
+### Github Oauth App:
+
+In GitHub, within your account Settings, click [Oauth Apps](https://github.com/settings/developers) under Developer Settings.
+
+click "New Oauth App".
+
+For the **Authorization callback URL**, enter the url for the "authorizing" page that you created above (e.g https://your-url/github/authorizing). Fill out the other fields with your custom values.
+_Note: If you are testing your app locally, you may need a separate development Github app (with a localhost redirect), and a production Github app._ 
+
+The generated **Client ID** & **Client Secret** will be consumed by the `createCreateAccessTokenFn` [defined above](#createcreateaccesstokenfn).
```

Comment:
Maybe add a bit to this section about how to use those variables. Specifically

1. How to add them to your environment variables 
2. How to register them in the `next.config.js`